### PR TITLE
update azure paths so that they work with no credentials

### PR DESCRIPTION
**Pull request by henktillman**

### CODE REVIEW

**Comment by ianand** [comment]
(thx, following, https://twitter.com/ianand/status/1767827185170300935)

**Comment by hanw** [comment]
Here is a unit test for the credential problem. It has something to do with `blobfile.listdir`

```
blobfile._common.Error: Encountered an error when requesting an access token: `invalid_request: AADSTS900144: The request body must contain the following parameter: 'client_id'. Trace ID: 0122581e-b632-44bc-a67d-10a375870d00 Correlation ID: 32846173-b7cc-4195-a763-1646b9860cf5 Timestamp: 2024-03-14 01:03:45Z`.  You can attempt to fix this by re-running `az login`.
```

```
import blobfile as bf

pieces_path = f"az://openaipublic/neuron-explainer/subject-models/gpt2/small/model_pieces"
piece_files = list(bf.listdir(pieces_path))
print(piece_files)
```

**Comment by swj0418** [comment]
When you try to access model files in browser, such as `gpt2"` via `https://openaipublic.blob.core.windows.net/neuron-explainer/subject-models/gpt2/small`, it says that the specified blob file does not exist. 

On the other hand, pre-computed activations `https://openaipublic.blob.core.windows.net/neuron-explainer/gpt2_small_data/collated-activations/{layer_index}/{neuron_index}.json` can be accessed via browser.

I think model paths are broken.

**Comment by VincentStark** [comment]
@henktillman you sir are my hero. I was about to start fixing it myself.

As a sidenote, I don't understand how this project can have 2.9k stars, and nobody took notice that it actually doesn't work for anyone outside of OAI.

**Comment by hanw** [comment]
Who decided to turn off the issue tab? How are we supposed to report issue without it? No one except people at OAI has permission to create a PR.

File: neuron_explainer/models/transformer.py [hunk 1]
```diff
@@ -697,12 +698,10 @@ def load(
         simplify: bool = False,
         simplify_kwargs: dict[str, Any] | None = None,
     ) -> "Transformer":
-        if bf.exists(name_or_path):
+        if name_or_path.startswith("https://"):
             path = name_or_path
         else:
-            path = f"az://openaipublic/neuron-explainer/subject-models/{name_or_path.replace('-', '/')}"
-        if not bf.exists(path):
```
Comment (henktillman) [inline neuron_explainer/models/transformer.py:704]: I couldn't find a good way to check if any files exist in a directory like `https://openaipublic.blob.core.windows.net/neuron-explainer/subject-models/`. hitting it using `urllib.request` returns 404, there is no xml data associated with the directory, etc.

File: neuron_explainer/models/transformer.py [hunk 2]
```diff
@@ -722,21 +721,15 @@ def save_checkpoint(
...
         for k, v in self.state_dict().items():
-            with bf.BlobFile(osp.join(pieces_path, f"{k}.pt"), "wb") as f:
+            with CustomFileHandler(osp.join(pieces_path, f"{k}.pt"), "wb") as f:
                 torch.save(v, f)
 
     def load_state_from_checkpoint(
         self, path: str, device: Device | None = None, dtype: torch.dtype | None = None
     ):
         pieces_path = osp.join(path, "model_pieces")
-        piece_files = list(bf.listdir(pieces_path))
```
Comment (henktillman) [inline neuron_explainer/models/transformer.py:732]: again, couldn't find a good way to list files in a directory when you only have the https link

**Review by stevenbills — APPROVED** [review]

**Pull request merged**